### PR TITLE
Choose roles during player setup: Backend

### PR DESCRIPTION
**Pull request by alalonde**
Ref #311

Includes the following schema changes:
- Added a `PlayerRole` enum table to store a whitelist of roles
- Added the accompanying discord role IDs to the metafam `guild_metadata` row
- Added a `player_roles` join table

Also includes a refactor where a bunch of player aggregation relationship names were cleaned up / standardized.

### CODE REVIEW

File: hasura/migrations/1630635244186_create_table_public_PlayerRole/up.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE "public"."PlayerRole"("role" text NOT NULL, "label" text NOT NULL, PRIMARY KEY ("role") , UNIQUE ("role"));
+
+INSERT INTO "public"."PlayerRole" (role, label) 
+VALUES ('PATRON', 'Patron'), ('BUILDER', 'Builder'), ('ARTIST', 'Artist'), ('SHILLER', 'Shiller'), ('INNKEEPER', 'Innkeeper'), ('DESIGNER', 'Designer'), ('WRITER', 'Writer');
```
Comment (mquellhorst) [inline hasura/migrations/1630635244186_create_table_public_PlayerRole/up.sql:4]: Any specific reason not all roles are in here?

As for the current guilds I'm missing:
Bughunter
Headhunter
XP-Fairy
Necromancer

As for other roles I'm missing:
Productizer
Rainmaker
Videolord
Soundmaster
Rogue
Bridgebuilder

File: hasura/migrations/1630635736784_add_player_discord_roles/up.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+UPDATE guild_metadata 
+SET discord_metadata = jsonb_set(discord_metadata, '{playerRoles}', '{"PATRON": "771329952691650600", "BUILDER": "764735856157655051", "ARTIST": "764735862461431808", "SHILLER": "764735870464557056", "INNKEEPER": "764736391937916949", "DESIGNER": "790300196771856444", "WRITER": "790300765641768971"}')
```
Comment (mquellhorst) [inline hasura/migrations/1630635736784_add_player_discord_roles/up.sql:2]: Same thing about the missing roles

File: packages/web/graphql/fragments.ts [hunk 3]
```diff
@@ -10,25 +10,25 @@ export const PlayerFragment = gql`
...
     }
-    playerType {
+    type {
       description
       id
       imageUrl
       title
     }
-    Player_Skills {
+    skills {
```
Comment (mquellhorst) [inline packages/web/graphql/fragments.ts:24]: I'm not sure, you made changes to hasura to reflect this right?
If so there are more references in: 

seed-local-db.js

File: packages/web/graphql/fragments.ts [hunk 4]
```diff
@@ -10,25 +10,25 @@ export const PlayerFragment = gql`
     ethereum_address
     availability_hours
     timezone
-    ColorAspect {
+    color_aspect {
```
Comment (mquellhorst) [inline packages/web/graphql/fragments.ts:13]: References in:

seed-local-db.js
graphql/getPersonalityInfo.ts

File: packages/web/graphql/fragments.ts [hunk 5]
```diff
@@ -10,25 +10,25 @@ export const PlayerFragment = gql`
...
       name
       description
       mask
     }
-    playerType {
+    type {
```
Comment (mquellhorst) [inline packages/web/graphql/fragments.ts:18]: References in:

seed-local-db.js

File: packages/backend/src/handlers/triggers/playerRoleChanged.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    const discordPlayer = await guild.members.fetch(playerDiscordId);
+    if (discordPlayer == null) {
+      console.warn(
+        `No discord player with ID ${playerDiscordId} found in server ${guild.name}!`,
+      );
+      return;
+    }
+
+    if (oldPlayerRole != null && newPlayerRole == null) {
```
Comment (mquellhorst) [inline packages/backend/src/handlers/triggers/playerRoleChanged.ts:65]: Looks good only question I'm not sure of: you sure this takes into account people having multiple roles?

**Review by mquellhorst — CHANGES_REQUESTED** [review]

File: hasura/migrations/1630635244186_create_table_public_PlayerRole/up.sql (hunk 1 above)
Comment (alalonde) [reply hasura/migrations/1630635244186_create_table_public_PlayerRole/up.sql:4]: Yes, this is intentional and addressed in #311.

File: packages/backend/src/handlers/triggers/playerRoleChanged.ts (hunk 6 above)
Comment (alalonde) [reply packages/backend/src/handlers/triggers/playerRoleChanged.ts:65]: This trigger gets called when a player_role record is inserted or deleted, so only for one role at a time.

File: packages/web/graphql/fragments.ts (hunk 3 above)
Comment (alalonde) [reply packages/web/graphql/fragments.ts:24]: The original change was in hasura, then I manually updated the various references to these types in the backend and frontend code.

File: packages/web/graphql/fragments.ts (hunk 5 above)
Comment (alalonde) [reply packages/web/graphql/fragments.ts:18]: Good catch; i'll fix that

File: packages/web/graphql/fragments.ts (hunk 4 above)
Comment (alalonde) [reply packages/web/graphql/fragments.ts:13]: In getPersonalityInfo it is still loading all the ColorAspects as opposed to that of the current player

**Comment by alalonde** [comment]
@mquellhorst Please re-review

**Review by mquellhorst — APPROVED** [review]
Looks good :D

**Pull request merged**